Cookbook/Proposals/ModelFixtures.md

```diff
@@ -0,0 +1,249 @@
+- "Mock" is misleading. Term "mock" is very overloaded, so let's set on this short definition first (["TestDouble", martinfowler.com](https://martinfowler.com/bliki/TestDouble.html)):
+
+> Mocks are pre-programmed with expectations which form a specification of the calls they are expected to receive. They can throw an exception if they receive a call they don't expect and are checked during verification to ensure they got all the calls they were expecting.
+
+It's clear that "mock" doesn't fit the issue we're dealing with in this proposal. Using "mock" term also may wrongly give the idea that returned value is a test double, while in fact it's a type used production.
+
+### Use Object Mother pattern to setup test data
+
+What's an Object Mother? To quote [Nat Pryce](http://natpryce.com/articles/000714.html):
+
+> An Object Mother is a class that contains a number of (usually static) Factory Methods that create objects for use in tests. For example, we could create an Object Mother for invoices we want to use in tests:
+ >
+> ```java
+> Invoice invoice = TestInvoices.newDeerstalkerAndCapeInvoice();
+> ```
+> An Object Mother helps keep tests readable by moving the code that creates new objects out of the tests themselves and giving clear names to the objects being constructed. It also helps maintain the test data by gathering the code that creates new objects together into the Object Mother class and allowing it to be reused between tests.
+
+This nicely fits the problem I'm aiming to fix with this proposal.
+
+Let's consider some points against this approach. In our codebase we already have something similar, e.g.
+
+```swift
+public enum MockPatientDetailsDTO {
+    public static func makeWithPartner() -> PatientDTO { ... }
+    public static func makeUpdated() -> PatientDTO { ... }
+    public static func makeMinor() -> PatientDTO { ... }
+    public static func makeEphemeralMinor() -> PatientDTO { ... }
+}
+```
+
+This kind of setup obscures what's going on and it's very hard to tell which data is changed to achieve end result. It's also impractical when we'd like to use similar data to one returned from existing helper, but with slight variation. We can't adjust the returned value, since our models have immutable properties. We could tweak data by adding parameters to those methods, but then we would end up with approach similar to proposed `fixture` helper. Model creation for tests would still be scattered across multiple places. Other solution would be to add another factory method on Object Mother, but this would get even less maintenable with time.
+
+If we encounter case where there are many instances of passing the same parameters to the `fixture` method, then we could consider adding methods like this on case by case basis, rather than making it a default.
```

Comment:
@gtsiap @diegopetrucci I guess this should answer your questions.

So we would have something like this:

```
extension SomeDTO {
    func makeSomeFixture() -> SomeDTO
    func makeAnotherFixture() -> SomeDTO
}
```